### PR TITLE
Add metadata to orders

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it adds metadata editor to order view.

**PR intended to be tested with API branch:** master

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [CU-7pqnkd Add metadata to order views](https://app.clickup.com/t/7pqnkd)

File: src/misc.ts [hunk 1]
```diff
@@ -235,7 +235,7 @@ export function hasErrors(errorList: UserError[] | null): boolean {
 export function getMutationState(
   called: boolean,
   loading: boolean,
-  ...errorList: UserError[][]
+  ...errorList: any[][]
```
Comment (orzechdev) [inline src/misc.ts:238]: Why any?

**Review by krzysztofwolski — APPROVED** [review]
I have the the same question as @orzechdev. Besides that - LGTM

File: src/misc.ts (hunk 1 above)
Comment (dominik-zeglen) [reply src/misc.ts:238]: Well, we only need info `how many` errors were returned, it's no use for us to know `what` errors

Comment (patrys) [reply src/misc.ts:238]: What do we gain from turning this into `any`? What if we wanted to explicitly forbid using `any` tomorrow to get stronger type checking?

**Review by AlicjaSzu — APPROVED** [review]

**Pull request merged**